### PR TITLE
Add Section block + Custom Feature Data block

### PR DESCRIPTION
**Pull request by KhadijaMahanga**
## Description

- [x] Add a gutenberg ChartsRow Block
- [x] Add section block that will use `ChartsRow block` as innerblocks
- [x] Add custom section block for featured data

Fixes [#172494877](https://www.pivotaltracker.com/story/show/172494877)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Desktop Screenshots
Peek 2020-04-23 17-51

Peek 2020-04-23 17-16

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: wp-content/plugins/hurumap/acf.php [hunk 1]
```diff
@@ -68,6 +68,18 @@ function filter_by_lang( $post ) {
     return get_posts(['numberposts' => 1, 'post_type' => $post->post_type, 'post__in' => [$post->ID], 'suppress_filters' => 0])[0];
 }
 
+if( function_exists('acf_add_options_page') ) {
```
Comment (kilemensi) [inline wp-content/plugins/hurumap/acf.php]: Guess we're not yet using this?

File: wp-content/plugins/hurumap/index.php [hunk 2]
```diff
@@ -330,14 +330,25 @@ function hurumap_data_blocks_register()
...
+        register_block_type('hurumap/section-chart', array(
+            'editor_script' => 'hurumap-data-blocks-script',
+        ));
+        register_block_type('hurumap/charts-row-block', array(
+            'editor_script' => 'hurumap-data-blocks-script',
+        ));
+
+        register_block_type('hurumap/section-block', array(
+            'editor_script' => 'hurumap-data-blocks-script',
+        ));
```
Comment (kilemensi) [inline wp-content/plugins/hurumap/index.php]: A bit documentation on these: Section Block -> Charts Row Block -> Section Chart? If so, may be `section-block`, `section-row-block` & `section-row-chart-block` would make this relation clear... Not sure though.

**Review by kilemensi — COMMENTED** [review]
:+1: @KhadijaMahanga ... Looks dope.

----------------------------------

1. I guess with this design, adding a chart in a third column can be quite cumbersome since the block is so small,
2. Lets use specific name rather than generic Section... May be HURUmap Section  (for the main block),
3. But my biggest concerns are:

    1. Why are we limit number of rows (as oppose to just accepting a number), and
    2. Is it possible to increase or reduce this number once created? Say we start today with just one row of demographics charts but we get new data and wants to add an additional row to it.

Otherwise, I haven't played with API side of it. Does this section block only work via hydration (meaning for us to use it on Country Profile pages were going to have to create those pages in the Dashboard) or can we pull it using the same `hurumap-data` API end point?

File: wp-content/plugins/hurumap/index.php (hunk 2 above)
Comment (KhadijaMahanga) [reply wp-content/plugins/hurumap/index.php]: @kilemensi yes, I will rename them

**Comment by KhadijaMahanga** [comment]
>  @KhadijaMahanga ... Looks dope.
> 
> 1. I guess with this design, adding a chart in a third column can be quite cumbersome since the block is so small,
> 2. Lets use specific name rather than generic Section... May be HURUmap Section  (for the main block),
> 3. But my biggest concerns are:
>    
>    1. Why are we limit number of rows (as oppose to just accepting a number), and
>    2. Is it possible to increase or reduce this number once created? Say we start today with just one row of demographics charts but we get new data and wants to add an additional row to it.
> 
> Otherwise, I haven't played with API side of it. Does this section block only work via hydration (meaning for us to use it on Country Profile pages were going to have to create those pages in the Dashboard) or can we pull it using the same `hurumap-data` API end point?

@kilemensi 
1. For charts, may be we can move inputs to the inspector control panel. What do you think?

3. From the meeting we had last time, the project owners wants short section. Thats the reason I have limited to 3, but I can change that too.

Also, I have created another story for the section work. https://www.pivotaltracker.com/story/show/172494897 This PR will deliver the featured data which we will use hydration for it

**Pull request merged**